flow/flow.go

```diff
@@ -46,11 +46,55 @@ var openFlowsNumber = uint32(0)
 var ringName = 1
 var createdPorts []port
 var schedState *scheduler
+var vEach [10][burstSize]uint8
+
+type processLine struct {
+	out      []*low.Ring
+	contexts []UserContext
+	scalar   bool
+}
 
 // Flow is an abstraction for connecting flow functions with each other.
 // Flow shouldn't be understood in any way beyond this.
 type Flow struct {
-	current *low.Ring
+	current  *low.Ring
+	line     *processLine
+	previous **Func
+}
+
+type partitionCtx struct {
+	currentAnswer       int
+	currentCompare      uint64
+	currentPacketNumber uint64
+	N                   uint64
+	M                   uint64
+}
+
+func (c partitionCtx) Copy() interface{} {
+	n := new(partitionCtx)
```

Comment:
Do you still hate struct literals?
return partitionCtx{N:c.N,M;c.M,currentCompare:c.N} is just 1 line...